Source/Modules/csharp.cxx

```diff
@@ -2044,7 +2044,7 @@ class CSHARP:public Language {
 	Printf(dcast_wrap->code, "  Swig::Director *director = (Swig::Director *) 0;\n");
 	Printf(dcast_wrap->code, "  jobject jresult = (jobject) 0;\n");
 	Printf(dcast_wrap->code, "  %s *obj = *((%s **)&jCPtrBase);\n", norm_name, norm_name);
-	Printf(dcast_wrap->code, "  if (obj) director = dynamic_cast<Swig::Director *>(obj);\n");
+	Printf(dcast_wrap->code, "  if (obj) director = static_cast<Swig::Director *>(obj);\n");
 	Printf(dcast_wrap->code, "  if (director) jresult = director->swig_get_self(jenv);\n");
```

Comment:
This looks good, but may be it should be a conditional compile time choice? Some people might want the run time safety. You could add a compile flag or check compiler specific variables like __GXX_RTTI.